src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1046,6 +1046,7 @@ bool QgsPostgresProvider::loadFields()
     mDefaultValues.insert( mAttributeFields.size(), defValMap[tableoid][attnum] );
 
     QgsField newField = QgsField( fieldName, fieldType, fieldTypeName, fieldSize, fieldPrec, fieldComment, fieldSubType );
+    newField.setArrayFormatString( "{%1}" );
```

Comment:
Shouldn't that be conditional for hstore fields?
And don't forget `QStringLiteral`

Reply:
array field has to be formatted this way too, so it works for all postgres field, no?

Reply:
Forget about my previous comment, I forget about the json fields